test/specs/components/Popup/Popup-test.tsx

```diff
@@ -36,6 +38,57 @@ describe('Popup', () => {
   }: PositionTestInput & { rtl?: never }) =>
     testPopupPosition({ align, position, expectedPlacement, rtl: true })
 
+  const getPopupContent = (popup: ReactWrapper) => {
+    return popup.find('.ui-popup__content')
+  }
+
+  const togglePopupTest = (keyName: string, keyboardKey: keyboardKey, onProp: PopupEvents) => {
+    test(`toggle popup with ${keyName} key`, () => {
+      const triggerId = 'triggerElement'
+      const popupContent = 'any test content'
+      const popup = mountWithProvider(
+        <Popup
+          trigger={<span id={triggerId}> text to trigger popup </span>}
+          content={popupContent}
+          on={onProp}
+        />,
+      )
+      const popupTriggerElement = popup.find(`#${triggerId}`)
+      popupTriggerElement.simulate('keydown', { keyCode: keyboardKey })
+      expect(getPopupContent(popup).length).toBe(1)
+      expect(getPopupContent(popup).getElement().props.children).toMatch(popupContent)
+
+      popupTriggerElement.simulate('keydown', { keyCode: keyboardKey })
+      expect(getPopupContent(popup).length).toBe(0)
+    })
+  }
+
+  const openAndClosePopupTest = (
+    keyNameToOpen: string,
+    keyboardKeyToOpen: keyboardKey,
+    keyboardKeyToClose: keyboardKey,
+    onProp: PopupEvents,
+  ) => {
+    test(`open popup with ${keyNameToOpen} key and close it with escape key`, () => {
+      const triggerId = 'triggerElement'
+      const popupContent = 'any test content'
+      const popup = mountWithProvider(
+        <Popup
+          trigger={<span id={triggerId}> text to trigger popup </span>}
```

Comment:
`text to trigger popup` can be safely omitted - it will be better to do that so the test's arrangement will be more concise